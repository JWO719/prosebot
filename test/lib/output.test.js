const Output = require('../../lib/output')

describe('output', () => {
  let map, output

  beforeEach(() => {
    map = new Map()
    const obj = {
      'filename.md': '# This is some text',
      'anotherfile.md': 'this **Has** some Problems. it sure does jason?',
      'cats.md': 'So the cat was stolen.'
    }

    for (const key in obj) {
      map.set(key, obj[key])
    }

    output = new Output(map)
  })

  describe('#removeMarkdownFromFiles', () => {
    it('removes markdown from the files and returns the expected map', () => {
      const actual = output.removeMarkdownFromFiles()
      expect(actual).toMatchSnapshot()
    })
  })

  describe('#buildWriteGoodResults', () => {
    it('returns the expected result', () => {
      const noMd = output.removeMarkdownFromFiles()
      const actual = output.buildWriteGoodResults(noMd)
      expect(actual).toMatchSnapshot()
    })
  })

  describe('#reachConclusion', () => {
    it('returns `failure` if there are any suggestions', () => {
      const noMd = output.removeMarkdownFromFiles()
      const results = output.buildWriteGoodResults(noMd)
      const actual = output.reachConclusion(results)
      expect(actual).toBe('failure')
    })

    it('returns `success` if there are no suggestions', () => {
      const actual = output.reachConclusion(new Map())
      expect(actual).toBe('success')
    })
  })

  describe('#buildSummary', () => {
    it('returns the expected result if the conclusion is `success`', () => {
      const actual = output.buildSummary('success')
      expect(actual).toBe('No issues have been found, great job!')
    })

    it('returns the expected string if there are suggestions', () => {
      const noMd = output.removeMarkdownFromFiles()
      const results = output.buildWriteGoodResults(noMd)
      const actual = output.buildSummary('failure', results)
      expect(actual).toMatchSnapshot()
    })
  })

  describe('#buildAnnotations', () => {
    it('returns the expected array of annotations', () => {
      const noMd = output.removeMarkdownFromFiles()
      const results = output.buildWriteGoodResults(noMd)
      const actual = output.buildAnnotations(noMd, results)
      expect(actual).toMatchSnapshot()
    })
  })
})
