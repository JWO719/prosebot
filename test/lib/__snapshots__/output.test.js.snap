// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`output #buildAnnotations returns the expected array of annotations 1`] = `
Array [
  Object {
    "annotation_level": "warning",
    "end_line": 1,
    "message": "\\"So\\" adds no meaning",
    "path": "cats.md",
    "start_line": 1,
  },
  Object {
    "annotation_level": "warning",
    "end_line": 1,
    "message": "\\"was stolen\\" may be passive voice",
    "path": "cats.md",
    "start_line": 1,
  },
]
`;

exports[`output #buildSummary returns the expected string if there are suggestions 1`] = `"2 suggestions have been found in 1 file"`;

exports[`output #buildWriteGoodResults returns the expected result 1`] = `
Map {
  "filename.md" => Array [],
  "anotherfile.md" => Array [],
  "cats.md" => Array [
    Object {
      "index": 0,
      "offset": 2,
      "reason": "\\"So\\" adds no meaning",
    },
    Object {
      "index": 11,
      "offset": 10,
      "reason": "\\"was stolen\\" may be passive voice",
    },
  ],
}
`;

exports[`output #removeMarkdownFromFiles removes markdown from the files and returns the expected map 1`] = `
Map {
  "filename.md" => "This is some text",
  "anotherfile.md" => "this Has some Problems. it sure does jason?",
  "cats.md" => "So the cat was stolen.",
}
`;
